plugins/rbac-backend/src/role-manager/role-manager.ts

```diff
@@ -86,47 +98,29 @@ export class BackstageRoleManager implements RoleManager {
     }
 
     // name1 is always user in our case.
-    // name2 is user or group.
+    // name2 is role or group.
     // user(name1) couldn't inherit user(name2).
     // We can use this fact for optimization.
     const { kind } = parseEntityRef(name2);
     if (kind.toLocaleLowerCase() === 'user') {
       return false;
     }
 
-    const memo = new AncestorSearchMemo(
-      name1,
-      this.tokenManager,
-      this.catalogApi,
-      this.catalogDBClient,
-    );
-    await memo.buildUserGraph(memo);
+    if (await this.cacheResults(name1)) {
+      const cachedResult = this.roleCache?.get(name1)!;
 
-    memo.debugNodesAndEdges(this.log, name1);
-    if (!memo.isAcyclic()) {
-      const cycles = memo.findCycles();
+      return cachedResult.has(name2);
+    }
 
-      this.log.warn(
-        `Detected cycle dependencies in the Group graph: ${JSON.stringify(
-          cycles,
-        )}. Admin/(catalog owner) have to fix it to make RBAC permission evaluation correct for groups: ${JSON.stringify(
-          cycles,
-        )}`,
-      );
+    // If the cache does not exist we need to build the graph and use it
+    // Otherwise we need to use the cache instead
+    const memo = await this.buildGraph(name1);
 
+    if (!this.detectCycleError(memo, name1)) {
```

Comment:
This code checks cycle dependencies only on build cache, but doesn't when cache exists.

Reply:
No the check is still happening whenever we cache, it instead will log the error and not cache the results. See this [line](https://github.com/janus-idp/backstage-plugins/blob/303801f63bf81f15eb4a67326d98655d1111b1e2/plugins/rbac-backend/src/role-manager/role-manager.ts#L284). This will still allow us to log the warning and we will return false if the role was not cached. 

The only downside to this, that I am just now noticing, is that the graph will be built a second time and the warning will be logged again.